mlbusinesscomponents/src/main/java/com/mercadolibre/android/mlbusinesscomponents/components/touchpoint/view/flex_cover_carousel/FlexCoverCarouselComponent.kt

```diff
@@ -0,0 +1,98 @@
+import com.mercadolibre.android.mlbusinesscomponents.components.utils.ScaleUtils
+
+class FlexCoverCarouselComponent(
+    context: Context,
+    attr: AttributeSet?
+) : RecyclerView(context, attr), FlexCoverCarouselComponentInterface {
+
+    private val verticalPadding = resources.getDimensionPixelSize(R.dimen.ui_1_5m)
+    private val linearLayoutManager = LinearLayoutManager(context, HORIZONTAL, false)
+    private val presenter = FlexCoverCarouselComponentPresenter(this)
+    private var cardsAdapter: FlexCoverCarouselComponentAdapter? = null
+    private val snapHelper = PagerSnapHelper()
+    private var isInitialized = false
+    private var wasAtFirstElement: Boolean? = null
+    private var wasAtLastElement: Boolean? = null
+    private var normalPadding = 0
+    private var biggerPadding = 0
+
+    override fun bind(model: FlexCoverCarouselResponse, onClickCallback: OnClickCallback?) {
+        if (!isInitialized) {
+            initialize()
+        }
+        presenter.bind(model, onClickCallback)
+    }
+
+    override fun notifyPadding(additionalEdgeInsets: AdditionalEdgeInsets) {
+        normalPadding = ScaleUtils.getPxFromDp(context, additionalEdgeInsets.left.toFloat()).toInt()
+        biggerPadding = ScaleUtils.getPxFromDp(context, additionalEdgeInsets.right.toFloat()).toInt()
+    }
+
+    override fun setCards(cards: List<FlexCoverCard>, onClickCallback: OnClickCallback?) {
+        cardsAdapter?.let {
+            cardsAdapter?.updateData(cards)
+        } ?: let {
+            cardsAdapter = FlexCoverCarouselComponentAdapter(cards, onClickCallback)
+            adapter = cardsAdapter
+            val decorator =
+                FlexCoverCarouselComponentDecorator(resources.getDimensionPixelSize(R.dimen.ui_1m))
+            addItemDecoration(decorator)
+        }
+        updatePadding(cards.size, wasAtFirstElement == null || isAtFirstElement(0), isAtLastElement(0))
+    }
+
+    private fun initialize() {
+        layoutManager = linearLayoutManager
+        clipToPadding = false
+        clipChildren = false
+        snapHelper.attachToRecyclerView(this)
+        addOnScrollListener(createScrollListener())
+    }
+
+    private fun createScrollListener() = object : OnScrollListener() {
+
+        override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {}
+
+        override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+            super.onScrolled(recyclerView, dx, dy)
+            updatePadding(cardsAdapter?.itemCount ?: 0, isAtFirstElement(dx), isAtLastElement(dx))
+        }
+    }
+
+    private fun updatePadding(itemCount: Int, isAtFirstElement: Boolean, isAtLastElement: Boolean) {
+        if (isAtFirstElement == wasAtFirstElement || isAtLastElement == wasAtLastElement) return
+        val (leftPadding, rightPadding) = when {
+            itemCount <= 1 -> normalPadding  to normalPadding // normal = 16, bigger = 104
```

Comment:
Esto quedó comentado 😅 
```suggestion
            itemCount <= 1 -> normalPadding  to normalPadding
```